server.py

```diff
@@ -192,25 +210,27 @@ def createSession(username):
 def login():
     username = bottle.request.forms.get("username")
     password = bottle.request.forms.get("password")
-    with shelve.open(user_db) as users:
-        if not username in users:
-            return bottle.template("home.html", message="User does not exist.")
-        if users[username].password != password:
-            return bottle.template("home.html", message="Incorrect password.")
+    if (
+        len(
+            User.select().where(
+                (User.username == username) & (User.password == password)
+            )
+        )
+        == 0
+    ):
+        return bottle.template("home.html", message="Invalid credentials.")
     return createSession(username)
 
 
 @app.post("/register")
 def register():
     username = bottle.request.forms.get("username")
     password = bottle.request.forms.get("password")
-    with shelve.open(user_db) as users:
-        if username in users:
-            return bottle.template(
-                "home.html",
-                message="Username already exists. Select a different username",
-            )
-        users[username] = User(password=password)
+    if len(User.select().where(User.username == username)) > 0:
```

Comment:
a better way of expressing this logic:

```python
try:
    user = User.create(username=username, password=password)
except IntegrityError:
    return "exists already"
```

since in the model definition you already have username = unique